service/history/execution/workflow_execution_util.go

```diff
@@ -47,5 +54,19 @@ func TerminateWorkflow(
 		terminateDetails,
 		terminateIdentity,
 	)
+
+	domainName := mutableState.GetDomainEntry().GetInfo().Name
```

Comment:
I suggest moving these metrics to AddWorkflowExecutionTerminatedEvent. It will simplify things since mutableStateBuilder already has metrics.Client and logger. Then you don't need to change the signature of this method